src/EdFi.AnalyticsMiddleTier.DataStandard32/Ews/PostgreSQL/0007-View-StudentSectionGradeFact-Alter.sql

```diff
@@ -0,0 +1,113 @@
+﻿-- SPDX-License-Identifier: Apache-2.0
+-- Licensed to the Ed-Fi Alliance under one or more agreements.
+-- The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+-- See the LICENSE and NOTICES files in the project root for more information.
+CREATE OR REPLACE VIEW analytics.ews_StudentSectionGradeFact
+AS
+    SELECT
+        Student.StudentUniqueId AS StudentKey,
+        CAST(Grade.SchoolId AS VARCHAR) AS SchoolKey,
+        CONCAT(Grade.GradingPeriodDescriptorId, '-', Grade.SchoolId, '-', to_char(GradingPeriod.BeginDate, 'yyyymmdd'))
+            AS GradingPeriodKey,
+        CONCAT(Student.StudentUniqueId, '-', Grade.SchoolId, '-', Grade.LocalCourseCode, '-', Grade.SchoolYear, '-',
+            Grade.SectionIdentifier, '-', Grade.SessionName, '-', to_char(Grade.BeginDate, 'yyyymmdd')) AS StudentSectionKey,
+        CONCAT(Grade.SchoolId, '-', Grade.LocalCourseCode, '-', Grade.SchoolYear, '-', Grade.SectionIdentifier, '-', 
+            Grade.SessionName) AS SectionKey,
+        COALESCE(Grade.NumericGradeEarned, ews_LetterGradeTranslation.NumericGradeEarned) AS NumericGradeEarned,
+        Grade.LetterGradeEarned,
+        GradeType.CodeValue AS GradeType
+    FROM
+        edfi.Grade
+    INNER JOIN
+        edfi.Student
+    ON
+        Grade.StudentUSI = edfi.Student.StudentUSI
+    INNER JOIN
+        edfi.Descriptor AS GradeType
+    ON
+        Grade.GradeTypeDescriptorId = GradeType.DescriptorId
+    INNER JOIN
+        edfi.GradingPeriod
+    ON
+        Grade.GradingPeriodDescriptorId = GradingPeriod.GradingPeriodDescriptorId
+    AND
+        Grade.GradingPeriodSequence = GradingPeriod.PeriodSequence
+    AND
+        Grade.SchoolId = GradingPeriod.SchoolId
+    AND
+        Grade.GradingPeriodSchoolYear = GradingPeriod.SchoolYear
+    INNER JOIN
+        edfi.Descriptor AS GradingPeriodDescriptor
+    ON
+        GradingPeriod.GradingPeriodDescriptorId = GradingPeriodDescriptor.DescriptorId
+    INNER JOIN
+        analytics_config.DescriptorMap
+    ON
+        GradeType.DescriptorId = DescriptorMap.DescriptorId
+    INNER JOIN
+        analytics_config.DescriptorConstant
+    ON
+        DescriptorConstant.DescriptorConstantId = DescriptorMap.DescriptorConstantId
+    LEFT OUTER JOIN
+        analytics_config.ews_LetterGradeTranslation
+    ON
+        Grade.LetterGradeEarned = ews_LetterGradeTranslation.LetterGradeEarned
+    WHERE
+        DescriptorConstant.ConstantName IN('GradeType.GradingPeriod', 'GradeType.Semester', 'GradeType.Final');
+
+IF EXISTS
```

Comment:
I think there is a copy-paste issue in this file.

Reply:
Done.